src/main.py

```diff
@@ -32,10 +34,29 @@ def calculate_scaled_mouse_position(position: tuple[int, int]) -> tuple[int, int
 
         self.app_state_manager.ui_manager.calculate_scaled_mouse_position = calculate_scaled_mouse_position
 
+    def run_debug(self) -> None:
+        surface = virtual_screen
+        font = pygame.font.SysFont("Comic Sans", 18)
+        black = (0, 0, 0)
+        current_state_name = self.app_state_manager.active_state.name if self.app_state_manager.active_state else "?"
+        previous_state_name = (
+            self.app_state_manager.previous_state.name if self.app_state_manager.previous_state else "?"
+        )
+        state_display = font.render(
+            f"Game State: {current_state_name}\n" f"Previous State: {previous_state_name}",
+            True,
+            black,
+        )
+        surface.blit(state_display, (10, 10))
```

Comment:
Nit: since this variable is only used once it would be slightly less cognitive burden to use it directly:

```diff
-         surface = virtual_screen
```

```suggestion
        virtual_screen.blit(state_display, (10, 10))
```